File: DSA/Java/tests/dsa/container/lists/queues/MyQueueTest.java
package dsa.container.lists.queues;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.*;


public class MyQueueTest {
    private MyQueue<Integer> queue;

    @Before
    public void setUp() {
        queue = new MyQueue<>();
    }

    @Test
    public void emptyQueue() {
        assertTrue(this.queue.empty());
    }

    @Test
    public void pushNotEmpty() {
        this.queue.push(1);
        assertFalse(this.queue.empty());
    }

    @Test
    public void popEmpty() {
        this.queue.push(1);
        assertFalse(this.queue.empty());

        this.queue.pop();
        assertTrue(this.queue.empty());
    }

    @Test
    public void pushPop() {
        this.queue.push(1);
        this.queue.push(2);
        assertEquals(Integer.valueOf(1), this.queue.pop());
    }

    @Test
    public void peek() {
        this.queue.push(1);
        this.queue.push(2);
        assertEquals(Integer.valueOf(1), this.queue.peek());
    }

    @Test
    public void front() {
        this.queue.push(1);
        this.queue.push(2);
        assertEquals(Integer.valueOf(1), this.queue.front());
    }

    @Test
    public void sizeEmpty() {
        assertEquals(0, this.queue.size());
    }

    @Test
    public void sizePush() {
        this.queue.push(1);
        assertEquals(1, this.queue.size());
    }

    @Test
    public void sizePop() {
        this.queue.push(1);
        this.queue.push(2);
        assertEquals(2, this.queue.size());

        assertEquals(Integer.valueOf(1), this.queue.pop());
        assertEquals(1, this.queue.size());
    }
}

File: DSA/Cpp/test/dsa/container/lists/queues/my_queue_test.cpp
#include <dsa/container/lists/queues/my_queue.hpp>
#include "queue_test.h"

namespace dsa::container::lists::queues {
namespace {

class MyQueueTest : public QueueTest {
 protected:
    MyQueue<int> queue;
};
TEST_F(MyQueueTest, emptyQueue) {
    EXPECT_TRUE(this->queue.empty());
}
TEST_F(MyQueueTest, pushNotEmpty) {
    this->queue.push(1);
    EXPECT_FALSE(this->queue.empty());
}
TEST_F(MyQueueTest, pushPopEmpty) {
    this->queue.push(1);
    EXPECT_EQ(1, queue.pop());
    EXPECT_TRUE(this->queue.empty());
}
TEST_F(MyQueueTest, pop) {
    this->queue.push(1);
    this->queue.push(2);
    EXPECT_EQ(1, queue.pop());
}
TEST_F(MyQueueTest, sizeEmpty) {
    EXPECT_EQ(0, queue.size());
}
TEST_F(MyQueueTest, pushSize) {
    queue.push(1);
    EXPECT_EQ(1, queue.size());
}
TEST_F(MyQueueTest, peek) {
    queue.push(1);
    queue.push(2);
    EXPECT_EQ(1, queue.peek());
}
TEST_F(MyQueueTest, front) {
    queue.push(1);
    queue.push(2);
    EXPECT_EQ(1, queue.front());
}

}
}
